### PR TITLE
[DeadCode] Using SideEffectNodeDetector on RemoveUnusedVariableAssignRector

### PR DESCRIPTION
**Pull request by samsonasik**
on is used detection.

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/DeadCode/Rector/Assign/RemoveUnusedVariableAssignRector.php [hunk 1]
```diff
@@ -147,7 +146,7 @@ private function isUsed(Assign $assign, Variable $variable): bool
 
         /** @var FuncCall|MethodCall|New_|NullsafeMethodCall|StaticCall $expr */
         $expr = $assign->expr;
-        if (! $this->isCall($expr)) {
+        if (! $this->sideEffectNodeDetector->detectCallExpr($expr)) {
```
Comment (Bl00D4NGEL) [inline rules/DeadCode/Rector/Assign/RemoveUnusedVariableAssignRector.php:149]: Could this also be done in [\Rector\DeadCode\Rector\Assign\RemoveDoubleAssignRector::isCall](https://github.com/rectorphp/rector-src/blob/13d73b380d5adedbccbe37afb0aff2e2319a7e27/rules/DeadCode/Rector/Assign/RemoveDoubleAssignRector.php#L95) ?

Comment (Bl00D4NGEL) [reply rules/DeadCode/Rector/Assign/RemoveUnusedVariableAssignRector.php:149]: Nevermind, didn't see https://github.com/rectorphp/rector-src/pull/679

**Comment by TomasVotruba** [comment]
Thank you 😌

**Pull request merged**